packages/core/content-manager/server/controllers/relations.js

```diff
@@ -186,39 +186,43 @@ module.exports = {
     }
 
     const queryParams = {
-      sort: mainField,
-      ...query,
-      fields: fieldsToSelect, // cannot select other fields as the user may not have the permissions
-      filters: {}, // cannot filter for RBAC reasons
+      select: fieldsToSelect,
     };
 
     if (!isEmpty(idsToOmit)) {
-      addFiltersClause(queryParams, { id: { $notIn: idsToOmit } });
-    }
-
-    // searching should be allowed only on mainField for permission reasons
-    if (_q) {
-      addFiltersClause(queryParams, { [mainField]: { $containsi: _q } });
+      queryParams.filters = { id: { $notIn: idsToOmit } };
     }
 
-    const subQuery = strapi.db.queryBuilder(sourceModel.uid);
-
-    const alias = subQuery.getAlias();
-
-    const knexSubQuery = subQuery
-      .where({ id, [`${alias}.id`]: { $notNull: true } })
-      .join({ alias, targetField })
-      .select(`${alias}.id`)
-      .getKnexQuery();
-
-    addFiltersClause(queryParams, { id: { $in: knexSubQuery } });
-
     if (MANY_RELATIONS.includes(attribute.relation)) {
-      ctx.body = await strapi.entityService.findPage(targetedModel.uid, queryParams);
+      const page = Number(query.page || 1);
+      const pageSize = Number(query.pageSize || 10);
+
+      queryParams.offset = Math.max(page - 1, 0) * pageSize;
+      queryParams.limit = pageSize;
+
+      const [results, count] = await Promise.all([
+        strapi.db
+          .query(sourceModelUid)
+          .load({ id }, targetField, { ...queryParams, ordering: 'desc' }),
+        strapi.db.query(sourceModelUid).load({ id }, targetField, { ...queryParams, count: true }),
+      ]);
```

Comment:
So now we directly use the Entity Manager and not the Entity Service.

Based on what logic should we use the EntityService or the entity manager ? I understood the flow was always this one 
![image](https://user-images.githubusercontent.com/20578351/193543381-0deebc19-4baf-4d70-a3f5-d486e4ed9dd9.png)

Reply:
Yes you're right indeed, I'll modify that!